src/ovp_pipeline/synthesis/contradiction_crystal.py

```
@@ -322,8 +322,27 @@ def _crystal_filename(contradiction_id: str) -> str:
 
 
+def _related_notes_section(slugs: tuple[str, ...]) -> str:
+    """Machine-generated ``## 相关笔记`` section appended to every
+    contradiction crystal.  Same rationale as the community crystal
+    helper — keeps the source-note backlink graph deterministic
+    instead of relying on the LLM to consistently emit ``[[slug]]``
+    inside prose.
+    """
+    lines = ["## 相关笔记", ""]
+    for slug in slugs:
+        lines.append(f"- [[{slug}]]")
```

Ruff (0.15.12)

[warning] Use `list.extend` to create a transformed list

(PERF401)